File: NetworkCheck/network_check.py
import platform
import subprocess

def check_network():
    os_type = platform.system()
    
    if os_type == "Windows":
        command = ["ping", "8.8.8.8", "-n", "1"]
    else:
        command = ["ping", "8.8.8.8", "-c", "1"]
    
    try:
        output = subprocess.check_output(command, stderr=subprocess.STDOUT)
        print("Сеть доступна")
    except subprocess.CalledProcessError:
        print("Сеть недоступна")

if __name__ == "__main__":
    check_network()

File: NetworkCheck/network_check.cpp
#include <iostream>
#include <cstdlib>
#include <string>

void check_network() {
    std::string os_type;
#ifdef _WIN32
    os_type = "Windows";
#else
    os_type = "Unix";
#endif

    std::string command;
    if (os_type == "Windows") {
        command = "ping 8.8.8.8 -n 10";
    } else {
        command = "ping 8.8.8.8 -c 10";
    }

    int result = std::system(command.c_str());
    if (result == 0) {
        std::cout << "Network is available" << std::endl;
    } else {
        std::cout << "Network is unavailable" << std::endl;
    }
}

int main() {
    check_network();
    std::cout << "Press Enter to exit..." << std::endl;
    std::cin.get();
    return 0;
}